test/ProductConstructionService.DependencyFlow.Tests/PendingUpdatesTests.cs

```diff
@@ -57,4 +57,32 @@ public async Task PendingUpdatesUpdatablePr()
             AndShouldHavePullRequestCheckReminder();
         }
     }
+
+    [Test]
+    public async Task PendingUpdatesNotUpdatableGroupingTest()
+    {
+        GivenATestChannel();
+        GivenASubscription(
+            new SubscriptionPolicy
+            {
+                Batchable = true,
+                UpdateFrequency = UpdateFrequency.EveryBuild
+            });
+        Build b1 = GivenANewBuild(true);
+        b1.Commit = "sha1";
+        Build b2 = GivenANewBuild(true);
+        b2.Commit = "sha2";
+        Build b3 = GivenANewBuild(true);
+        b3.Commit = "sha3";
+
+        using (WithExistingCodeFlowPullRequest(b1, canUpdate: false))
+        {
+            await WhenProcessPendingUpdatesAsyncIsCalled(b2);
+            await WhenProcessPendingUpdatesAsyncIsCalled(b3);
+
+            ThenShouldHaveInProgressPullRequestState(b1, b3.Commit);
```

Comment:
This sounds wrong. The PR state should be `b3`?

Reply:
no, because the PR is not updatable, so both b2 and b3 didn't apply, but we want to apply b3 next.
I see now that I'm missing a test where this scenario has already happened, the PR is updatable, and the cache is in this state. We try to process an update with b2, fail because it wants b3, and then b3 succeeds

Reply:
Oh, I see